### ISSUE TITLE
Generated declaration file for constructor nullable parameter property is non-nullable

### ISSUE THREAD

**Issue by patsissons** [Microsoft/TypeScript#15872]
**TypeScript Version:**  2.2.2

**Code**

```ts
// typescript code
export class Foo {
  constructor(public bar?: string) {
  }
}
```
```ts
// generated declaration
export declare class Foo {
    bar: string;
    constructor(bar?: string);
}
```

**Expected behavior:**
`Foo.bar` should be typed as `string | undefined`

**Actual behavior:**
`Foo.bar` is typed as `string`

Our declaration is no longer compatible with the source that generated it.

**Comment by NaridaL** [comment]
Also affects 2.4.0.

**Comment by sandersn** [comment]
Fix is up at #16497
